frontend/components/FileDetails/FileDetails.tsx

```diff
@@ -58,32 +62,49 @@ const FileDetails = ({
           )}
         </div>
       </div>
-      {!progress && canEdit && onFileSelect && (
-        <GitOpsModeTooltipWrapper
-          position="left"
-          tipOffset={-8}
-          renderChildren={(disableChildren) => (
-            <div className={`${baseClass}__edit`}>
-              <Button
-                disabled={disableChildren}
-                className={`${baseClass}__edit-button`}
-                variant="icon"
-              >
-                <label htmlFor="edit-file">
-                  <Icon name="pencil" color="ui-fleet-black-75" />
-                </label>
-              </Button>
-              <input
-                disabled={disableChildren}
-                accept={accept}
-                id="edit-file"
-                type="file"
-                onChange={onFileSelect}
-              />
-            </div>
-          )}
-        />
-      )}
+      {!progress &&
+        canEdit &&
+        onFileSelect &&
+        (gitopsCompatible ? (
+          <GitOpsModeTooltipWrapper
+            position="left"
+            tipOffset={4}
+            renderChildren={(disableChildren) => (
+              <div className={`${baseClass}__edit`}>
+                <Button
+                  disabled={disableChildren}
+                  className={`${baseClass}__edit-button`}
+                  variant="icon"
+                >
+                  <label htmlFor="edit-file">
+                    <Icon name="pencil" color="ui-fleet-black-75" />
+                  </label>
+                </Button>
+                <input
+                  disabled={disableChildren}
+                  accept={accept}
+                  id="edit-file"
+                  type="file"
+                  onChange={onFileSelect}
+                />
+              </div>
+            )}
+          />
+        ) : (
+          <div className={`${baseClass}__edit`}>
```

Comment:
What's the difference between this and starting at L74? Seems like those bits could be extracted into a const and then included in both places so they don't drift?

Reply:
The two `disabled={disableChildren}`s are the difference. The way `GitOpsModeTooltipWrapper`'s `renderChildren` works makes it slightly hairy to encapsulate the child which may or may not be disabled, making this kind of repetition a decent approach